mixxorz/slippers
# Adding support for multiline template tags.

**Description**
For Slippers, one thing became really important was the # of parameters used when reusing components, with just a few parameters we were way past [79/88 characters per line](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#:~:text=We%20allow%20up%20to%2088,f%2Dstrings%2C%20or%20str.) defined by its PEP-8.
This PR introduces a monkey-patch into Django' regex used to tokenize the tags internally, this isn't near great as a official solution but serves as an workaround to it.

I don't see a fix for this any time soon as mentioned here: https://github.com/django/django/pull/2556/ and https://github.com/django/deps/pull/3

**How to Reproduce**
Using Slippers codebase
1. Checkout into this branch
2. Run the tests: `python runtests.py`

Using your own codebase
1. Paste the following snippet inside the `settings.py`
```py
import re
from django.template import base
base.tag_re = re.compile(base.tag_re.pattern, re.DOTALL)
```
2. Open any of your template file, indent any tag used, as follow
```
{% static 'media/folder/other/folder/file.jpg' %}
```
to 
```
{%
  static
  'media/folder/other/folder/file.jpg'
%}
```
3. Assuming you did patched the `tag_re`, everything should work as before with no side-effects

**Motivation**
Before this PR, things got a bit wild when using components:
```
{% #button aria-role="button" x-bind:class="!open ? 'hidden': ''" @click="myHandler" %}Hello{% /button %}
```
And now we could do things like:
```
{%
  #button
  aria-role="button"
  x-bind:class="!open ? 'hidden': ''"
  @click="myHandler"
%}
  Hello
{% /button %}
```
